### PR TITLE
feat(v2): DR volume

### PR DESCRIPTION
**Pull request by c3y1huang**
#### Which issue(s) this PR fixes:

Issue longhorn/longhorn#6613

#### What this PR does / why we need it:

Support v2 volume for incremental restore.

#### Special notes for your reviewer:

`None`

#### Additional documentation or context

`None`

### CODE REVIEW

**Comment by derekbit** [comment]
I forgot if I've implemented the details that are required for v2 DR volume in the go-spdk-helper side. Don't we need to update go-spdk-helper?

**Comment by DamiaSan** [comment]
It seems good to me but i don't have so much experience on Longhorn backup and restore to understand all the things involved, better to wait for Shuo and Derek to approve.

**Review by shuo-wu — DISMISSED** [review]

File: deltablock.go [hunk 1]
```diff
@@ -933,10 +934,16 @@ func RestoreDeltaBlockBackupIncrementally(ctx context.Context, config *DeltaRest
 		return err
 	}
 	go func() {
-		defer volDev.Close()
+		var err error
+		finalProgress := 0
+
 		defer func() {
+			_ = deltaOps.CloseVolumeDev(volDev)
```
Comment (derekbit) [inline deltablock.go:941]: Can we log the error message?

File: deltablock.go [hunk 2]
```diff
@@ -894,14 +895,14 @@ func RestoreDeltaBlockBackupIncrementally(ctx context.Context, config *DeltaRest
...
+	volDev, volDevPath, err := deltaOps.OpenVolumeDev(volDevName)
 	if err != nil {
-		return err
+		return errors.Wrapf(err, "failed to open volume device %v", volDevName)
 	}
 	defer func() {
 		// make sure to close the device
 		if err != nil {
-			_ = volDev.Close()
+			_ = deltaOps.CloseVolumeDev(volDev)
```
Comment (derekbit) [inline deltablock.go:905]: Can we log the error message?

File: deltablock.go (hunk 1 above)
Comment (c3y1huang) [reply deltablock.go]: Fixed, thank you!

**Review by derekbit — APPROVED** [review]
LGTM

**Pull request merged**